python/tests/expressions/test_visitors.py

```diff
@@ -339,61 +327,61 @@ def test_always_false_or_always_true_expression_binding(table_schema_simple: Sch
     [
         (
             And(
-                In(Reference("foo"), (literal("foo"), literal("bar"))),
-                In(Reference("bar"), (literal(1), literal(2), literal(3))),
+                In(Reference("foo"), {"foo", "bar"}),
+                In(Reference("bar"), {1, 2, 3}),
             ),
             And(
-                BoundIn[str](
+                BoundIn(
```

Comment:
I was able to fix the mypy failure for this expression by reverting the changes on line 339 and adding the `[str]` type param to `BoundReference`.

```
error: Cannot infer type argument 1 of "BoundIn"
```

We don't want or expect people to create bound expressions by hand, so it's fine to require `Iterable[Literal[T]]` and require that the reference matches. Binding handles this already, so this can be a little unfriendly.